Userland/wc.cpp

```diff
@@ -2,143 +2,131 @@
+void wc_out(WhatToPrint& flags, Count& count, String name)
 {
-    Count total { "total" };
-    for (const auto& c : counts) {
-        report(c);
-        total.lines += c.lines;
-        total.words += c.words;
-        total.chars += c.chars;
-    }
-    if (counts.size() > 1) {
-        report(total);
-    }
-    fflush(stdout);
-}
+    if (flags.line_flag)
+        printf("%7i ", count.lines);
+    if (flags.word_flag)
+        printf("%7i ", count.words);
+    if (flags.byte_flag)
+        printf("%7lu ", count.bytes);
+    if (flags.char_flag)
+        printf("%7i ", count.characters);
 
-int count_words(const char* s)
-{
-    int n = 0;
-    bool in_word = false;
-    for (; *s; ++s) {
-        if (!isspace(*s)) {
-            if (!in_word) {
-                in_word = true;
-                ++n;
-            }
-        } else if (in_word) {
-            in_word = false;
-        }
-    }
-    return n;
+    printf("%14s\n", name.characters());
 }
 
-int main(int argc, char** argv)
+int main(int argc, char **argv)
 {
-    if (argc < 2) {
-        printf("usage: wc [-c|-m] [-lw] [file...]\n");
-        return 0;
-    }
+    struct stat st;
+    int current_character;
+    int iteration = 0;
+    FILE *file_pointer = stdin;
+    WhatToPrint arg_flags;
+    Word word_flags;
+    Count single_count;
+    Count total_count;
 
     CArgsParser args_parser("wc");
-    args_parser.add_arg("l", "Include lines in count");
-    args_parser.add_arg("c", "Include bytes in count");
-    args_parser.add_arg("m", "Include chars in count");
-    args_parser.add_arg("w", "Include words in count");
-    CArgsParserResult args = args_parser.parse(argc, (char**)argv);
-
+    args_parser.add_arg("l", "Output line count");
+    args_parser.add_arg("c", "Output byte count");
+    args_parser.add_arg("m", "Output character count");
+    args_parser.add_arg("w", "Output word count");
+    CArgsParserResult args = args_parser.parse(argc, argv);
     if (args.is_present("l")) {
-        output_lines = true;
+        arg_flags.line_flag=true;
     }
     if (args.is_present("c")) {
-        output_chars = true;
+        arg_flags.byte_flag=true;
     }
     if (args.is_present("m")) {
-        output_chars = true;
+        arg_flags.char_flag=true;
     }
     if (args.is_present("w")) {
-        output_words = true;
-    }
-    if (!output_lines && !output_words && !output_chars) {
-        output_lines = output_chars = output_words = true;
-    }
-
-    Vector<String> files = args.get_single_values();
-    Vector<Count> counts;
-    int status;
-    if (files.is_empty()) {
-        status = wc("", counts);
-        if (status != 0)
-            return status;
-    } else {
-        for (const auto& f : files) {
-            status = wc(f, counts);
-            if (status != 0)
-                return status;
+        arg_flags.word_flag=true;
+    }
+    if (!arg_flags.line_flag && !arg_flags.char_flag && !arg_flags.word_flag && !arg_flags.byte_flag) {
```

Comment:
This is made very hard-to-read due to the unnecessary grouping of the flag variables in a struct.